tests/endpoints/mint.rs

```diff
@@ -0,0 +1,495 @@
+        10u64,
+        ITHEUM_TOKEN_IDENTIFIER,
+        0u64,
+        10u64 + 100u64,
+        Some(TxExpect::user_error("str:Contract not ready")),
+    );
+
+    state.bond_contract_default_deploy_and_set(10u64, 100u64);
+
+    state.minter_mint(
+        FIRST_USER_ADDRESS_EXPR,
+        "Test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "random-url-encoded-here",
+        "https://test.com/test",
+        1000u64,
+        5u64,
+        &"Test title".repeat(1),
+        &"Test description".repeat(1),
+        10u64,
+        ITHEUM_TOKEN_IDENTIFIER,
+        0u64,
+        100u64,
+        None,
+    );
+
+    state.world.check_state_step(
+        CheckStateStep::new()
+            .put_account(
+                FIRST_USER_ADDRESS_EXPR,
+                CheckAccount::new().esdt_balance(ITHEUM_TOKEN_IDENTIFIER_EXPR, "100"), // 200-100 = 100
+            )
+            .put_account(
+                MINTER_CONTRACT_ADDRESS_EXPR,
+                CheckAccount::new().esdt_balance(ITHEUM_TOKEN_IDENTIFIER_EXPR, ""),
+            )
+            .put_account(
+                BONDING_CONTRACT_ADDRESS_EXPR,
+                CheckAccount::new().esdt_balance(ITHEUM_TOKEN_IDENTIFIER_EXPR, "100"), // 100 for BOND
+            ),
+    );
+}
+
+#[test]
+fn mint_with_anti_spam_tax_test_and_whitelist() {
+    let mut state = ContractsState::new();
+    let first_user_address = state.first_user.clone();
+
+    state
+        .mock_minter_initialized(ITHEUM_TOKEN_IDENTIFIER, 100u64, 10u64)
+        .unpause_minter_contract(MINTER_OWNER_ADDRESS_EXPR, None)
+        .bond_contract_default_deploy_and_set(10u64, 100u64)
+        .bond_unpause_contract(BONDING_OWNER_ADDRESS_EXPR, None);
+
+    state.minter_enable_whitelist(MINTER_OWNER_ADDRESS_EXPR, None);
+
+    state
+        .world
+        .set_state_step(SetStateStep::new().block_timestamp(11u64));
+
+    state.minter_add_to_whitelist(MINTER_OWNER_ADDRESS_EXPR, first_user_address.clone(), None);
+
+    state.minter_mint(
+        FIRST_USER_ADDRESS_EXPR,
+        "Test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "random-url-encoded-here",
+        "https://test.com/test",
+        1000u64,
+        5u64,
+        &"Test title".repeat(1),
+        &"Test description".repeat(1),
+        10u64,
+        ITHEUM_TOKEN_IDENTIFIER,
+        0u64,
+        100u64,
+        Some(TxExpect::user_error("str:Not enough funds")),
+    );
+
+    state.minter_mint(
+        FIRST_USER_ADDRESS_EXPR,
+        "Test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "https://test.com/test",
+        "random-url-encoded-here",
+        "https://test.com/test",
+        1000u64,
+        5u64,
+        &"Test title".repeat(1),
+        &"Test description".repeat(1),
+        10u64,
+        ITHEUM_TOKEN_IDENTIFIER,
+        0u64,
+        100u64 + 100u64,
```

Comment:
Shouldn't you have tests that check this?
require!(
payment.amount >= &price + &bond_amount,
ERR_NOT_ENOUGH_FUNDS
);

You are only checking if all is well when sending equal or lower funds. What if anti spam tax is 0 and you send equal AND more funds (separate cases). What if anti spam tax is not 0 and you send equal AND more funds (separate cases)?

Reply:
It should be 
```suggestion
require!(
payment.amount == &price + &bond_amount,
ERR_NOT_ENOUGH_FUNDS
);
```

Reply:
The check you are asking is in the same file under this test `mint_test_without_anti_spam_tax_test`

Reply:
Cool then, my bad